runtime/near-vm-runner/src/wasmer1_runner.rs

```diff
@@ -94,19 +96,57 @@ impl IntoVMError for wasmer::RuntimeError {
         let trap_code = self.clone().to_trap();
         match self.downcast::<VMLogicError>() {
-            Ok(e) => (&e).into(),
-            _ => {
-                if let Some(trap_code) = trap_code {
-                    // A trap
-                    VMError::FunctionCallError(FunctionCallError::Wasmer1Trap(
-                        trap_code.to_string(),
-                    ))
-                } else {
-                    // A general error
-                    VMError::FunctionCallError(FunctionCallError::WasmerRuntimeError(error_msg))
-                }
-            }
+            Ok(e) => return (&e).into(),
+            _ => {}
         }
+        let error = match trap_code {
+            Some(TrapCode::StackOverflow) => FunctionCallError::WasmTrap(WasmTrap::StackOverflow),
+            Some(TrapCode::HeapSetterOutOfBounds) => {
+                FunctionCallError::WasmTrap(WasmTrap::MemoryOutOfBounds)
+            }
+            Some(TrapCode::HeapAccessOutOfBounds) => {
+                FunctionCallError::WasmTrap(WasmTrap::MemoryOutOfBounds)
+            }
+            Some(TrapCode::HeapMisaligned) => {
+                FunctionCallError::WasmTrap(WasmTrap::MisalignedAtomicAccess)
+            }
+            Some(TrapCode::TableSetterOutOfBounds) => {
+                FunctionCallError::WasmTrap(WasmTrap::MemoryOutOfBounds)
+            }
+            Some(TrapCode::TableAccessOutOfBounds) => {
+                FunctionCallError::WasmTrap(WasmTrap::MemoryOutOfBounds)
+            }
+            Some(TrapCode::OutOfBounds) => FunctionCallError::WasmTrap(WasmTrap::MemoryOutOfBounds),
+            Some(TrapCode::IndirectCallToNull) => {
+                FunctionCallError::WasmTrap(WasmTrap::IndirectCallToNull)
+            }
+            Some(TrapCode::BadSignature) => {
+                FunctionCallError::WasmTrap(WasmTrap::IncorrectCallIndirectSignature)
+            }
+            Some(TrapCode::IntegerOverflow) => {
+                FunctionCallError::WasmTrap(WasmTrap::IllegalArithmetic)
+            }
+            Some(TrapCode::IntegerDivisionByZero) => {
+                FunctionCallError::WasmTrap(WasmTrap::IllegalArithmetic)
+            }
+            Some(TrapCode::BadConversionToInteger) => {
+                FunctionCallError::WasmTrap(WasmTrap::IllegalArithmetic)
+            }
+            Some(TrapCode::UnreachableCodeReached) => {
+                FunctionCallError::WasmTrap(WasmTrap::Unreachable)
+            }
+            Some(TrapCode::UnalignedAtomic) => {
+                FunctionCallError::WasmTrap(WasmTrap::MisalignedAtomicAccess)
+            }
+            Some(TrapCode::Interrupt) => {
+                FunctionCallError::Nondeterministic("Wasmer interrupt".to_string())
+            }
+            Some(TrapCode::VMOutOfMemory) => {
+                FunctionCallError::Nondeterministic("Wasmer out of memory".to_string())
+            }
+            None => panic!("Unknown error: {}", error_msg),
```

Comment:
nit: I'd pull that out out the match:

```
let trap_code = trap_code.unwrap_or_else(|| panic!("Unknown error: {}", error_msg))
```

No need to wrap each arm into Some that way.

Reply:
Done.